tests/gold_tests/connect/replays/connect.replay.yaml

```diff
@@ -0,0 +1,53 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+#
+# This replay file executes a CONNECT request and a subsequent GET request
+#
+meta:
+  version: "1.0"
+
+sessions:
+  - transactions:
+      - client-request:
+          method: CONNECT
+          version: "1.1"
+          url: www.example.com:80
+          headers:
+            fields:
+              - [uuid, 1]
+              - [Host, www.example.com:80]
+        # This is a CONNECT request so it should not reach the origin server
+        server-response:
+          status: 200
+
+        # ATS returns a 200 responses to client when it establishes a tunnel
+        # between the client and server
+        proxy-response:
+          status: 200
+
+      # Once the tunnel between client and server is established, subsequent
+      # requests will reach the server(via the tunnel)
+      - client-request:
+          method: GET
+          version: "1.1"
+          url: /get
+          headers:
+            fields:
+              - [uuid, 2]
+              - [Host, www.example.com]
+        server-response:
+          status: 200
```

Comment:
Let's add some `X-Response` header for this proxied transaction and add a `proxy-response` verification that the client gets that back. The content will be opaque to ATS, but the proxy-verifier server should still function as normal and the client should get its response back.

Reply:
Good point! Added those